utils/utils.go

```diff
@@ -411,6 +411,9 @@ func PostUploadProcessFile(serverFileName string, fileName string, tmpFile *os.F
 
 	if err != nil {
 		log.Println("Error extractiong thumbnail for video " + fileName)
+		if err := updateUploadStatus(db, fileName); err != nil {
+			log.Printf("Error updating upload status for video %s in DB: %v", fileName, err)
+		}
```

Comment:
I think all the functions called in the `PostUploadProcessFile` function will need a little bit of refactoring.

Change the return type for all the subfunctions you see and make it return an error as well.

then add checks for the error and wherever the error is returned, call your updateUploadStatus function.

in this flow, only thumbnail extraction should be optional. if thumbnail is extraction is errored out, it should gracefully log it and move on ahead in the function.